tests/reftests/repository-update.test

```diff
@@ -0,0 +1,181 @@
+N0REP0
+### <REPO/repo>
+opam-version: "2.0"
+### <REPO/packages/foo/foo.1/opam>
+opam-version: "2.0"
+synopsis: "Initial package"
+### <REPO/packages/bar/bar.1/opam>
+opam-version: "2.0"
+synopsis: "Another package"
+### <REPO/packages/baz/baz.1/opam>
+opam-version: "2.0"
+synopsis: "Third package"
+### opam switch create repo-update --empty
+### opam repository add default ./REPO --this-switch
+[default] synchronised from file://${BASEDIR}/REPO
+### opam list --all --all-versions -s
+bar.1
+baz.1
+foo.1
+### : Simple file creation
+### <REPO/packages/newpkg/newpkg.1/opam>
+opam-version: "2.0"
+synopsis: "New package created"
+### opam update default
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[default] synchronised from file://${BASEDIR}/REPO
+Now run 'opam upgrade' to apply any package updates.
+### opam list --all --all-versions -s
+bar.1
+baz.1
+foo.1
+newpkg.1
+### : File deletion
+### rm REPO/packages/bar/bar.1/opam
+### opam update default
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[default] synchronised from file://${BASEDIR}/REPO
+Now run 'opam upgrade' to apply any package updates.
+### opam list --all --all-versions -s
+baz.1
+foo.1
+newpkg.1
+### : File edit
+### <REPO/packages/foo/foo.1/opam>
+opam-version: "2.0"
+synopsis: "Updated package"
+### opam update default
```

Comment:
it would be nice if the new incremental behaviour was shown in the test. We could select the right OPAMDEBUGSECTION to show that only a limited number of files are being read